File: src/main/java/com/intellij/tapestry/intellij/toolwindow/nodes/MessageCatalogNode.java
package com.intellij.tapestry.intellij.toolwindow.nodes;

import com.intellij.tapestry.core.model.presentation.PresentationLibraryElement;
import com.intellij.tapestry.core.resource.IResource;

import javax.swing.tree.DefaultMutableTreeNode;

public class MessageCatalogNode extends DefaultMutableTreeNode {

    public MessageCatalogNode(Object userObject) {
        super(userObject);

        for (IResource catalog : ((PresentationLibraryElement) userObject).getMessageCatalog()) {
            add(new ResourceLeafNode(catalog));
        }
    }

    public String toString() {
        return "Message Catalog";
    }
}

File: src/main/kotlin/com/github/rar91279/plugin/tapestry/intellij/toolwindow/nodes/MessageCatalogNode.kt
package com.github.rar91279.plugin.tapestry.intellij.toolwindow.nodes

import com.github.rar91279.plugin.tapestry.core.model.presentation.PresentationLibraryElement
import javax.swing.tree.DefaultMutableTreeNode

class MessageCatalogNode(element: PresentationLibraryElement) : DefaultMutableTreeNode(element) {

    init {
        for (catalog in element.messageCatalog) add(ResourceLeafNode(catalog))
    }

    override fun toString() = "Message Catalog"
}
